yt_dlp/extractor/vidyard.py

```diff
@@ -0,0 +1,256 @@
+import re
+
+from .common import InfoExtractor
+from ..utils import (
+    extract_attributes,
+    int_or_none,
+    traverse_obj,
+)
+
+
+class VidyardBaseIE(InfoExtractor):
+
+    def _get_formats_and_subtitles(self, video_source, video_id):
+        video_source = video_source or {}
+        formats, subtitles = [], {}
+        for key, value in video_source.items():
+            if key == 'hls':
+                for video_hls in value:
+                    fmts, subs = self._extract_m3u8_formats_and_subtitles(video_hls.get('url'), video_id, headers={
+                        'referer': 'https://play.vidyard.com/',
+                    })
+                    formats.extend(fmts)
+                    self._merge_subtitles(subs, target=subtitles)
+            else:
+                formats.extend({
+                    'url': video_mp4.get('url'),
+                    'ext': 'mp4',
+                } for video_mp4 in value)
+
+        return formats, subtitles
+
+    def _get_direct_subtitles(self, caption_json):
+        subs = {}
+        for caption in caption_json:
+            subs.setdefault(caption.get('language') or 'und', []).append({
+                'url': caption.get('vttUrl'),
+                'name': caption.get('name'),
+            })
+
+        return subs
+
+    def _fetch_video_json(self, video_uuid, video_id=None):
+        video_id = video_id or video_uuid
+        return self._download_json(
+            f'https://play.vidyard.com/player/{video_uuid}.json', video_id)['payload']
+
+    def _process_video_json(self, json_data, video_id, **kwargs):
+        formats, subtitles = self._get_formats_and_subtitles(json_data['sources'], video_id)
+        self._merge_subtitles(self._get_direct_subtitles(json_data.get('captions')), target=subtitles)
+
+        return {
+            **kwargs,
```

Comment:
If applying the suggestion below:
```suggestion
    def _process_video_json(self, json_data, video_id):
        formats, subtitles = self._get_formats_and_subtitles(json_data['sources'], video_id)
        self._merge_subtitles(self._get_direct_subtitles(json_data.get('captions')), target=subtitles)

        return {
```

Reply:
Referrer handling extracted as suggested as of b8345dd2076859b85857a82e010d4b793978b041